var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1201,3 +751,435 @@ def test(self):
+        else:
+            config_args.append("--with-dbmliborder=")
+
+        if "+pyexpat" in spec:
+            config_args.append("--with-system-expat")
+        else:
+            config_args.append("--without-system-expat")
+
+        if "+ctypes" in spec:
+            config_args.append("--with-system-ffi")
+        else:
+            config_args.append("--without-system-ffi")
+
+        if "+tkinter" in spec:
+            config_args.extend(
+                [
+                    "--with-tcltk-includes=-I{0} -I{1}".format(
+                        spec["tcl"].prefix.include, spec["tk"].prefix.include
+                    ),
+                    "--with-tcltk-libs={0} {1}".format(
+                        spec["tcl"].libs.ld_flags, spec["tk"].libs.ld_flags
+                    ),
+                ]
+            )
+
+        # https://docs.python.org/3.8/library/sqlite3.html#f1
+        if spec.satisfies("+sqlite3 ^sqlite+dynamic_extensions"):
+            config_args.append("--enable-loadable-sqlite-extensions")
+
+        if spec.satisfies("%oneapi"):
+            cflags.append("-fp-model=strict")
+
+        if cflags:
+            config_args.append("CFLAGS={0}".format(" ".join(cflags)))
+
+        return config_args
+
+    def install(self, pkg, spec, prefix):
+        with working_dir(self.build_directory):
+            make(*self.install_targets, parllel=False)
+
+
+class GenericBuilder(GenericBuilder, RunAfter, BuildEnvironment):
+    phases = ("build", "install")
+
+    @property
+    def plat_arch(self):
+        """
+        String referencing platform architecture
+        filtered through Python's Windows build file
+        architecture support map
+
+        Note: This function really only makes
+        sense to use on Windows, could be overridden to
+        cross compile however.
+        """
+
+        arch_map = {"AMD64": "x64", "x86": "Win32", "IA64": "Win32", "EM64T": "Win32"}
+        arch = platform.machine()
+        if arch in arch_map:
+            arch = arch_map[arch]
+        return arch
+
+    @property
+    def win_build_params(self):
+        """
+        Arguments must be passed to the Python build batch script
+        in order to configure it to spec and system.
+        A number of these toggle optional MSBuild Projects
+        directly corresponding to the python support of the same
+        name.
+        """
+        args = []
+        args.append("-p %s" % self.plat_arch)
+        if self.spec.satisfies("+debug"):
+            args.append("-d")
+        if self.spec.satisfies("~ctypes"):
+            args.append("--no-ctypes")
+        if self.spec.satisfies("~ssl"):
+            args.append("--no-ssl")
+        if self.spec.satisfies("~tkinter"):
+            args.append("--no-tkinter")
+        return args
+
+    def build(self, pkg, spec, prefix):
+        """Makes the build targets specified by
+        :py:attr:``~.AutotoolsPackage.build_targets``
+        """
```

Comment:
This comment doesn't seem correct (refers to `AutotoolsPackage`):
```suggestion
```

Reply:
Yeah, I'm not sure what happened with the comments... The merge conflicts were fairly messy.